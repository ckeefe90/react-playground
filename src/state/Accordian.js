import React from 'react'

export default class Accordian extends React.Component {
    static defaultProps = { sections : [] };
    state = { currentSectionIndex: 0 };
    handleButtonClick(index) {
        this.setState({ currentSectionIndex: index })
    }
    render() {
        return (
        <div>
            <ul>
                {this.props.sections.map((section, i) => 
                <li key={i}>
                    <button className='title' onClick={() => this.handleButtonClick(i)}>{section.title}</button>
                    {this.state.currentSectionIndex === i && <p>{section.content}</p>}
                </li>)}
            </ul>
        </div>
        )
    }
}
