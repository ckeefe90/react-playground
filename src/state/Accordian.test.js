import React from 'react'
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordian from './Accordian.js'

describe(`Accordian Component`, () => {
    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
      ]
    it('renders without error', () => {
        const wrapper = shallow(<Accordian/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('renders no section by default', () => {
        const wrapper = shallow(<Accordian sections={sections}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('renders open clicked section', () => {
        const wrapper = shallow(<Accordian sections={sections}/>)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it('only opens one section at a time', () => {
        const wrapper = shallow(<Accordian sections={sections}/>)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})