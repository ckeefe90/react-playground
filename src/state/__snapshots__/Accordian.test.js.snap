// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component only opens one section at a time 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 3
      </button>
      <p>
        Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
      </p>
    </li>
  </ul>
</div>
`;

exports[`Accordian Component renders no section by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </p>
    </li>
    <li
      key="1"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordian Component renders open clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        className="title"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordian Component renders without error 1`] = `
<div>
  <ul />
</div>
`;
